### PR TITLE
Feat figma integration

### PR DESCRIPTION
**Pull request by ivan-015**

### CODE REVIEW

File: lib/input/figma/entities/layers/ellipse.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  factory FigmaEllipse.fromJson(Map<String, dynamic> json) =>
+      _$FigmaEllipseFromJson(json);
+  @override
+  Map<String, dynamic> toJson() => _$FigmaEllipseToJson(this);
+
+  @override
+  Future<PBIntermediateNode> interpretNode(PBContext currentContext) async {
+    imageReference = addToImageQueue(UUID);
+
+    // if (fillsList != null &&
```
Comment (siliconivan) [inline lib/input/figma/entities/layers/ellipse.dart]: Do we leave this in? @x64Eddie

File: lib/input/figma/entities/layers/frame.dart [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        verticalPadding: verticalPadding,
+        itemSpacing: itemSpacing,
+        flow: null,
+        children: children,
+        UUID: UUID,
+        backgroundColor: backgroundColor,
+      );
+
+      return Future.value(tempGroup.interpretNode(currentContext));
+      // PBBorder border;
```
Comment (siliconivan) [inline lib/input/figma/entities/layers/frame.dart]: Do we leave this?

File: lib/controllers/figma_controller.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  FigmaNodeTree generateFigmaTree(var jsonFigma, var projectname) {
+    try {
+      return FigmaNodeTree(projectname, jsonFigma);
+    } catch (e, stackTrace) {
+      print(e);
+      return null;
+    }
+  }
+
+  FigmaNodeTree declareScaffolds(FigmaNodeTree tree) {
```
Comment (x64Eddie) [inline lib/controllers/figma_controller.dart:46]: We might consider adding docs on why we have to explicitly declare what Figma elements are Scaffolds.

File: lib/input/figma/entities/layers/ellipse.dart (hunk 1 above)
Comment (x64Eddie) [reply lib/input/figma/entities/layers/ellipse.dart]: Nope

File: lib/input/figma/helper/style_extractor.dart [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      return FigmaColor(
+        alpha: json['a'],
+        blue: json['b'],
+        green: json['g'],
+        red: json['r'],
+      );
+    } else {
+      // Set color default to transparent if fill is null
+      return null;
+      // return FigmaColor(
```
Comment (x64Eddie) [inline lib/input/figma/helper/style_extractor.dart]: Lets remove this comment. 🤔

File: lib/input/sketch/entities/style/style.dart [hunk 5]
```diff
@@ -30,17 +39,52 @@ class Style {
...
+
+  @override
+  void set borders(List _borders) {
+    // TODO: implement borders
+  }
+
+  @override
+  Future<PBIntermediateNode> interpretNode(PBContext currentContext) {
+    // TODO: implement interpretNode
+    throw UnimplementedError();
```
Comment (x64Eddie) [inline lib/input/sketch/entities/style/style.dart]: This would cause a Runtime Exception if the style comes from Sketch, no? 🤔

**Review by x64Eddie — CHANGES_REQUESTED** [review]
Removal of commented out code and double check the Unimplemented functions in the Sketch Style class.

**Review by x64Eddie — CHANGES_REQUESTED** [review]
Removal of commented out code and double check the Unimplemented functions in the Sketch Style class.

File: lib/input/sketch/entities/style/style.dart (hunk 5 above)
Comment (SushiRoll53) [reply lib/input/sketch/entities/style/style.dart]: You are right

**Review by SushiRoll53 — APPROVED** [review]
LGTM!

**Review by x64Eddie — APPROVED** [review]
LGTM

**Pull request merged**